app/parsers/bulkrax/csv_parser.rb

```diff
@@ -10,19 +10,38 @@ def self.parser_fields
       }
     end
 
-    def run
-      create_works
-    end
-
     def records(_opts = {})
-      CSV.foreach(
+      @records ||= CSV.foreach(
         parser_fields['csv_path'],
         headers: true,
         header_converters: :symbol,
         encoding: 'utf-8'
       )
     end
 
+    def create_collections
+      return unless records.headers.include?(:collection)
+
+      acquire_lock_for(importer.id) do
+        records.each do |_collections|
+          next if record[:collection].blank?
+
+          # split by : ; |
+          collection.split(/\s*[:;|]\s*/).each do |collection|
+            metadata = {
+              title: [c],
+              Bulkrax.system_identifier_field => [collection],
+              visibility: 'open',
+              collection_type_gid: Hyrax::CollectionType.find_or_create_default_collection_type.gid
+            }
+
+            new_entry = collection_entry_class.where(importer: importer, identifier: collection, raw_metadata: metadata).first_or_create!
+            ImportWorkCollectionJob.perform_later(new_entry.id, importer.current_importer_run.id)
```

Comment:
I think for this lock to be affective these still need to be a perform_now?

Reply:
otherwise are they spinning off in the background while you blaze through creating works?  Specifically what happens in the 10, 50 or 100 worker cases?

Reply:
Yes, this is where I was struggling to test effectively, because my solr kept crashing. I'll spend some more time on this.